firestore/integration_test_internal/src/aggregate_query_test.cc

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firebase/firestore.h"
+#include "firestore_integration_test.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+
+using AggrgegateQueryTest = FirestoreIntegrationTest;
+
+size_t AggregateQueryHash(const AggregateQuery& aggregate_query) {
```

Comment:
Please add an inline comment about this `AggregateQueryHash` function and why it exists. It _appears_ that it exists because it's testing the private Hash() method from the `AggregateQuery` hash and is declared as a "friend" to get access to it. Also, please include in this comment _why_ you are testing a private method, since classes should normally be tested using their public interface.

This comment also applies to aggregate_query_snapshot_test.cc

Reply:
Hmmm... Not sure I can answer all that since I was basically following the pattern set forth by: https://github.com/firebase/firebase-cpp-sdk/blob/619e38cc071f4a31033924e30008e1d1e6e2483c/firestore/integration_test_internal/src/query_snapshot_test.cc

I am guessing this is an integration point with Unity, and therefore this helps catch problems before they bubble up to Unity. But I honestly don't have the big picture yet.

Reply:
Yes, I believe that you are right. Our existing testing coverage is not adequate, IMO. You won't find many other places where we test move/copy constructors. But that's very, very risky IMO since those methods are very nuanced. What I'd actually recommend is to just omit Hash() for now and add it in later once you have a concrete use case (i.e. when you start working on the unity sdk).